spec/models/hw_info_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+require 'rails_helper'
+
+describe HwInfo do
+  subject(:hw_info) { create(:system, :with_hw_info).hw_info }
+
+  let(:valid_uuid) { 'aaaaaaaa-aaaa-4aaa-9aaa-aaaaaaaaaaaa' }
+
+  it { is_expected.to belong_to :system }
+  it { is_expected.to validate_presence_of :system }
+
+  it 'enforces uniqueness' do
+    expect(hw_info).to validate_uniqueness_of(:system)
+  end
+
+  describe '.uuid' do
+    it { is_expected.to allow_value(nil).for(:uuid) }
+    it { is_expected.to allow_value('xyzzy').for(:uuid) }
+
+    it 'forces invalid uuid to nil' do
+      hw_info.uuid = 'xyzzy'
+      hw_info.save!
+
+      expect(hw_info.uuid).to be nil
+    end
+
+    it_behaves_like 'model with UUID format validation and nil forcing', :uuid
+
+    it 'has case-insensitive validations' do
```

Comment:
Try to make it clear that this is really a non-duplicate uuid issue.

You're not really testing the validation of the uuid format here.